src/main/java/assets/game.js

```diff
@@ -236,25 +240,58 @@ function place(size) {
         let row = this.parentNode.rowIndex;
         let col = this.cellIndex;
         vertical = document.getElementById("is_vertical").checked;
+        if (shipType == "SUBMARINE"){
+            submerged = document.getElementById("submerge").checked;
+        } else {
+            submerged = 0;
+        }
         let table = document.getElementById("player");
-        for (let i=0; i<size; i++) {
-            let cell;
-            if(vertical) {
-                let tableRow = table.rows[row+i];
-                if (tableRow === undefined) {
-                    // ship is over the edge; let the back end deal with it
-                    break;
+        if(shipType == "SUBMARINE"){
+                for (let i=0; i<size-1; i++) {
+                    let cell;
+                    if(vertical) {
+                        let tableRow = table.rows[row+i];
+                        if (tableRow === undefined) {
+                            // ship is over the edge; let the back end deal with it
+                            break;
+                        }
+                        cell = tableRow.cells[col];
+                    } else {
+                        cell = table.rows[row].cells[col+i];
+                    }
+                    if (cell === undefined) {
+                        // ship is over the edge; let the back end deal with it
+                        break;
+                    }
+                    cell.classList.toggle("placed");
```

Comment:
I see this is the cell listener for the submarine when hovering over the board.

Reply:
Yep!